presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TranslateExpressions.java

```diff
@@ -271,6 +292,213 @@ public Result apply(ValuesNode valuesNode, Captures captures, Context context)
+
+            if (!node.getStatisticsAggregation().isPresent()) {
+                return Result.empty();
+            }
+
+            Optional<StatisticAggregations> rewrittenStatisticsAggregation = cleanUpStatisticAggregation(node.getStatisticsAggregation().get(), context);
+
+            if (rewrittenStatisticsAggregation.isPresent()) {
+                return Result.ofPlanNode(new TableWriterNode(
+                        node.getId(),
+                        node.getSource(),
+                        node.getTarget(),
+                        node.getRowCountSymbol(),
+                        node.getFragmentSymbol(),
+                        node.getTableCommitContextSymbol(),
+                        node.getColumns(),
+                        node.getColumnNames(),
+                        node.getPartitioningScheme(),
+                        rewrittenStatisticsAggregation,
+                        node.getStatisticsAggregationDescriptor()));
+            }
+            return Result.empty();
+        }
+    }
+
+    private Optional<StatisticAggregations> cleanUpStatisticAggregation(StatisticAggregations statisticAggregations, Rule.Context context)
+    {
+        ImmutableMap.Builder<Symbol, AggregationNode.Aggregation> rewrittenAggregation = builder();
+        boolean changed = false;
+        for (Map.Entry<Symbol, AggregationNode.Aggregation> entry : statisticAggregations.getAggregations().entrySet()) {
+            AggregationNode.Aggregation rewritten = convertToRowExpression(entry.getValue(), context);
+            rewrittenAggregation.put(entry.getKey(), rewritten);
+            if (!rewritten.equals(entry.getValue())) {
+                changed = true;
+            }
+        }
+        if (changed) {
+            return Optional.of(new StatisticAggregations(rewrittenAggregation.build(), statisticAggregations.getGroupingSymbols()));
+        }
+        return Optional.empty();
+    }
+
+    private AggregationNode.Aggregation convertToRowExpression(AggregationNode.Aggregation aggregation, Rule.Context context)
+    {
+        Map<NodeRef<Expression>, Type> lambdaTypes = analyzeLambdaTypes(aggregation, context);
+
+        return new AggregationNode.Aggregation(
+                new CallExpression(
+                        aggregation.getCall().getDisplayName(),
+                        aggregation.getCall().getFunctionHandle(),
+                        aggregation.getCall().getType(),
+                        aggregation.getArguments().stream().map(argument -> removeOriginalExpression(argument, context, lambdaTypes)).collect(toImmutableList())),
+                aggregation.getFilter().map(filter -> removeOriginalExpression(filter, context)),
+                aggregation.getOrderBy(),
+                aggregation.isDistinct(),
+                aggregation.getMask());
+    }
+
+    private Map<NodeRef<Expression>, Type> analyzeLambdaTypes(AggregationNode.Aggregation aggregation, Rule.Context context)
```

Comment:
I think the class is complicated enough to have its own test

Reply:
Will add a test on translateAggregation